src/main/java/xyz/nkomarn/harbor/listener/AfkListener.java

```diff
@@ -0,0 +1,150 @@
+package xyz.nkomarn.harbor.listener;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+import org.bukkit.scheduler.BukkitRunnable;
+import xyz.nkomarn.harbor.Harbor;
+import xyz.nkomarn.harbor.provider.DefaultAFKProvider;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public final class AfkListener implements Listener {
+    private final DefaultAFKProvider afkProvider;
+    private Queue<AfkPlayer> players;
+    private PlayerMovementChecker movementChecker;
+
+    public AfkListener(DefaultAFKProvider afkProvider) {
+        this.afkProvider = afkProvider;
+        JavaPlugin.getPlugin(Harbor.class).getLogger().info("Initializing fallback AFK detection system. Fallback AFK system is not enabled at this time");
+    }
+
+    /**
+     * Provides a way to start the listener
+     */
+    public void start() {
+        JavaPlugin plugin = JavaPlugin.getPlugin(Harbor.class);
+        players = new ArrayDeque<>();
+        movementChecker = new PlayerMovementChecker();
+
+        // Populate the queue with any existing players
+        players.addAll(Bukkit.getOnlinePlayers().stream().map((Function<Player, AfkPlayer>) AfkPlayer::new).collect(Collectors.toSet()));
+
+        // Register listeners after populating the queue
+        Bukkit.getServer().getPluginManager().registerEvents(this, plugin);
+
+        // We want every player to get a check every 20 ticks. The runnable smooths out checking a certain
+        // percentage of players over all 20 ticks. Thusly, the runnable must run on every tick
+        movementChecker.runTaskTimer(plugin, 0, 1);
+
+        JavaPlugin.getPlugin(Harbor.class).getLogger().info("Fallback AFK detection system is enabled");
```

Comment:
Pass Harbor instance through the constructor

(ps you already had the harbor instance assigned to a variable in at the top of the method 🙃 )